gpudrive/datatypes/observation.py

```diff
@@ -366,3 +366,238 @@ def one_hot_encode_bev_map(self):
+
+        Returns:
+            A dict with the traffic light data for the current timestep
+        """
+        return {
+            "state": self.state[:, :, self.current_time],
+            "pos_x": self.pos_x[:, :, self.current_time],
+            "pos_y": self.pos_y[:, :, self.current_time],
+            "pos_z": self.pos_z[:, :, self.current_time],
+            "time_index": self.time_index[:, :, self.current_time],
+            "lane_id": self.lane_id[:, :, self.current_time],
+            "valid_mask": self.valid_mask[:, :, self.current_time],
+        }
+
+    def set_current_time(self, time_idx):
+        """Sets the current timestep for convenient access.
+
+        Args:
+            time_idx: The timestep index to use
+        """
+        max_time = self.state.shape[2] - 1
+        self.current_time = min(max(0, time_idx), max_time)
+
+    def get_stop_light_mask(self, time_idx=None):
+        """Returns a mask of traffic lights in the stop state.
+
+        Args:
+            time_idx: Optional specific timestep (defaults to current_time)
+
+        Returns:
+            A tensor with 1.0 for stop lights and 0.0 for others
+        """
+        time_idx = self.current_time if time_idx is None else time_idx
+        return (self.state[:, :, time_idx] == 1).float() * self.valid_mask[
+            :, :, time_idx
+        ]
+
+    def get_caution_light_mask(self, time_idx=None):
+        """Returns a mask of traffic lights in the caution state.
+
+        Args:
+            time_idx: Optional specific timestep (defaults to current_time)
+
+        Returns:
+            A tensor with 1.0 for caution lights and 0.0 for others
+        """
+        time_idx = self.current_time if time_idx is None else time_idx
+        return (self.state[:, :, time_idx] == 2).float() * self.valid_mask[
+            :, :, time_idx
+        ]
+
+    def get_go_light_mask(self, time_idx=None):
+        """Returns a mask of traffic lights in the go state.
+
+        Args:
+            time_idx: Optional specific timestep (defaults to current_time)
+
+        Returns:
+            A tensor with 1.0 for go lights and 0.0 for others
+        """
+        time_idx = self.current_time if time_idx is None else time_idx
+        return (self.state[:, :, time_idx] == 3).float() * self.valid_mask[
+            :, :, time_idx
+        ]
+
+    def predict_state_changes(self, future_window=10):
+        """Analyzes when traffic lights will change state within a future window.
+
+        Args:
+            future_window: Number of timesteps to look ahead
+
+        Returns:
+            Dictionary with time-to-change predictions for each traffic light
+        """
+        current_states = self.state[:, :, self.current_time]
+        time_to_change = torch.ones_like(current_states) * -1
+
+        # Look ahead to find when lights change
+        max_time = min(self.current_time + future_window, self.state.shape[2])
```

Comment:
this requires the caller to have set the `current_time` before, could be an unintended behavior if they have not